src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -141,6 +142,9 @@
     private static final List<String> RECOVERABLE_FAILURES = Arrays.asList(
             "Connection refused", "Connection reset", "Connection timed out", "No route to host", "Premature connection close"
     );
+    public static final Integer DEFAULT_MAX_NUM_RETRIES = 10;
```

Comment:
Wouldn't it make more sense to use `int` here? The member fields can be nullable, but these constants certainly aren't, and autoboxing will be performed regardless.

Reply:
I'm gonna check it because I have changed it due findbugs errors related to boxing-unboxing

Reply:
also, I will need an Integer when I refactor the constructors and use the DataBoundSetter methods

Reply:
That's the reason why i set the constants as Integer

At SSHLauncher.java:[line 1439] BX_UNBOXING_IMMEDIATELY_REBOXED

```
    public Integer getMaxNumRetries() {
        return maxNumRetries == null || maxNumRetries < 0 ? DEFAULT_MAX_NUM_RETRIES : maxNumRetries;
    }
```

I can change it to but it is ugly to transform something can be and Integer, I preferred to avoid the transformation.

```
    public Integer getMaxNumRetries() {
        return maxNumRetries == null || maxNumRetries < 0 ? Integer.valueOf(DEFAULT_MAX_NUM_RETRIES) : maxNumRetries;
    }
```

Reply:
Ah, ok, no worries. This would require changing far more signatures clearly; not worth it.